test/tests/jobs/GetJobTest.cpp

```diff
@@ -0,0 +1,408 @@
+    // Expect the jobs to be returned in order of low, medium, high, low, medium, high
+    void testPrioritiesWithDifferentNextRunTimes() {
+        // Create jobs of different priorities
+        // Low
+        SData command("CreateJob");
+        command["name"] = "low";
+        command["priority"] = "0";
+        command["firstRun"] = SComposeTime("%Y-%m-%d %H:%M:%S", STimeNow());
+        STable response = tester->executeWaitVerifyContentTable(command);
+        list<string> jobList;
+        jobList.push_back(response["jobID"]);
+
+        // Medium
+        command["name"] = "medium";
+        command["priority"] = "500";
+        command["firstRun"] = SComposeTime("%Y-%m-%d %H:%M:%S", STimeNow()+1000000);
+        response = tester->executeWaitVerifyContentTable(command);
+        jobList.push_back(response["jobID"]);
+
+        // High
+        command["name"] = "high";
+        command["priority"] = "1000";
+        command["firstRun"] = SComposeTime("%Y-%m-%d %H:%M:%S", STimeNow()+2000000);
+        response = tester->executeWaitVerifyContentTable(command);
+        jobList.push_back(response["jobID"]);
+
+        // High
+        command["name"] = "high";
+        command["priority"] = "1000";
+        command["firstRun"] = SComposeTime("%Y-%m-%d %H:%M:%S", STimeNow()+5000000);
+        response = tester->executeWaitVerifyContentTable(command);
+        jobList.push_back(response["jobID"]);
+
+        // Medium
+        command["name"] = "medium";
+        command["priority"] = "500";
+        command["firstRun"] = SComposeTime("%Y-%m-%d %H:%M:%S", STimeNow()+4000000);
+        response = tester->executeWaitVerifyContentTable(command);
+        jobList.push_back(response["jobID"]);
+
+        // Low
+        command["name"] = "low";
+        command["priority"] = "0";
+        command["firstRun"] = SComposeTime("%Y-%m-%d %H:%M:%S", STimeNow()+3000000);
+        response = tester->executeWaitVerifyContentTable(command);
+        jobList.push_back(response["jobID"]);
+
+        // Confirm these jobs all have different nextRun times
+        SQResult result;
+        tester->readDB("SELECT nextRun FROM jobs WHERE jobID IN (" + SComposeList(jobList) + ") GROUP BY nextRun;", result);
+        ASSERT_EQUAL(result.size(), 6);
+
+        // GetJob and confirm that the first 3 jobs are returned in order of low, medium, high
+        command.clear();
+        command.methodLine = "GetJob";
+        command["name"] = "*";
+        response = tester->executeWaitVerifyContentTable(command);
+        ASSERT_EQUAL(response["name"], "low");
+        sleep(1);
+        response = tester->executeWaitVerifyContentTable(command);
+        ASSERT_EQUAL(response["name"], "medium");
+        sleep(1);
+        response = tester->executeWaitVerifyContentTable(command);
+        ASSERT_EQUAL(response["name"], "high");
+
+        // Sleep 3 seconds so that the nextRun time for the last 'high' job is now
+        sleep(3);
```

Comment:
Either remove this and change the assertions below or change the comment in the description ;)
`// Expect the jobs to be returned in order of low, medium, high, low, medium, high`